apps/angular/router-input/src/app/test.component.ts

```diff
@@ -1,21 +1,18 @@
-import { AsyncPipe } from '@angular/common';
-import { Component, inject } from '@angular/core';
-import { ActivatedRoute } from '@angular/router';
-import { map } from 'rxjs';
+import { Component, Input } from '@angular/core';
+
 @Component({
   selector: 'app-test',
   standalone: true,
-  imports: [AsyncPipe],
+  imports: [],
   template: `
-    <div>TestId: {{ testId$ | async }}</div>
-    <div>Permission: {{ permission$ | async }}</div>
-    <div>User: {{ user$ | async }}</div>
+    <div>TestId: {{ testId }}</div>
+    <div>Permission: {{ permission }}</div>
+    <div>User: {{ user }}</div>
   `,
 })
 export default class TestComponent {
-  private activatedRoute = inject(ActivatedRoute);
-
-  testId$ = this.activatedRoute.params.pipe(map((p) => p['testId']));
-  permission$ = this.activatedRoute.data.pipe(map((d) => d['permission']));
-  user$ = this.activatedRoute.queryParams.pipe(map((q) => q['user']));
+  // Router Inputs
+  @Input() testId: string | undefined;
+  @Input() permission: string | undefined;
+  @Input() user: string | undefined;
```

Comment:
prefer to use the ? instead of undefined. Make the code cleaner. But that my personal preference

Reply:
ya that will be cleaner as well